### ISSUE TITLE
Multiline log assistance

### ISSUE THREAD

**Issue by danbeaulieu** [splunk/splunk-connect-for-kubernetes#134]
I have a Java service that logs on multiple lines. A sample when running `docker logs <containerid>`

```
[#|2019-04-17T21:42:02.279+0000|INFO|glassfish3.1.2|com.sun.jersey.api.container.filter.LoggingFilter|_ThreadID=42;_ThreadName=http-thread-pool-8080(1);|464 * Server in-bound request
 464 > GET http://192.168.3.4:8080/api/foo/bar
 464 > host: 192.168.3.4:8080
 464 > user-agent: kube-probe/1.13
 464 > accept-encoding: gzip
 464 > connection: close
 464 > 
 |#]
```
(There are more typical multiline application logs, but this was the most sanitized and easier to share. This shows up in stack traces as well, etc)

This is what it looks like in `/var/log/containers/<internalId>-myapp-6599bc5647-zclgs_<internalId>_myapp-fee0bbd21e2b3bfed5c6e25c4e47d8ea61eb7a6b53145dcc4f8a7064baea1043.log`:

```
{"log":"[#|2019-04-17T21:42:02.279+0000|INFO|glassfish3.1.2|com.sun.jersey.api.container.filter.LoggingFilter|_ThreadID=42;_ThreadName=http-thread-pool-8080(1);|464 * Server in-bound request\n","stream":"stderr","time":"2019-04-17T21:42:02.279277907Z"}
{"log":"464 \u003e GET http://192.168.3.4:8080/api/foo/bar\n","stream":"stderr","time":"2019-04-17T21:42:02.279310491Z"}
{"log":"464 \u003e host: 192.168.3.4:8080\n","stream":"stderr","time":"2019-04-17T21:42:02.279314859Z"}
{"log":"464 \u003e user-agent: kube-probe/1.13\n","stream":"stderr","time":"2019-04-17T21:42:02.279318139Z"}
{"log":"464 \u003e accept-encoding: gzip\n","stream":"stderr","time":"2019-04-17T21:42:02.279321489Z"}
{"log":"464 \u003e connection: close\n","stream":"stderr","time":"2019-04-17T21:42:02.279325661Z"}
{"log":"464 \u003e \n","stream":"stderr","time":"2019-04-17T21:42:02.27932941Z"}
{"log":"|#]\n","stream":"stderr","time":"2019-04-17T21:42:02.279333012Z"}
```
since the multiline start is essentially `[#|` I started with that. I added this to the override yaml file:

```
logs:
  myapp:
    from:
      pod: "*-myapp"
      container: myapp
    multiline: 
      firstline: /^\[#|/
```
This shows up in the configmap like this:

```
<filter tail.containers.var.log.containers.*-myapp*myapp*.log>
        @type concat
        key log
        timeout_label @SPLUNK
        stream_identity_key stream
        multiline_start_regexp /^\[#|/
        flush_interval 5s
      </filter>
```

And I confirmed this is what it looks like in the `/fluentd/etc/output.conf` file in the splunk-kubernetes-logging containers.

However in splunk all of these logs are still separate events. 

This is a k8s 1.13.5 cluster, using the 1.1.0 splunk/fluentd-hec image and 1.1.0 helm chart.
is this a config issue? An issue with the HEC?

**Comment by danbeaulieu** [comment]
This was user error. The pipe in the regex needed to be escaped.

**Comment by hovu96** [comment]
From my point of view, line breaking/merging should be done on Splunk side. I created [prototype](https://github.com/hovu96/fluent-plugin-splunk-hec/commit/02fff6814b19f720ce70126c483d5473d98ea17c) and there is also this [branch](https://github.com/splunk/fluent-plugin-splunk-hec/commit/fca9f4c368a523fedabab170af157bd8b0903cbd#diff-474f530618c2fd088becd7dc20da7ccd), both adding an option for using the RAW endpoint instead of the EVENT endpoint. I tested it and it worked for me. Many Splunk customers are asking for this. Anyone interested in working on a PR?
